backend/apis/leads/update-lead.api.js

```diff
@@ -83,6 +83,47 @@ app.patch("/api/leads/:id", (req, res, next) => {
       );
     }
 
+    const leadServicesChanged = atLeastOne(req.body, "leadServices");
+
+    if (leadServicesChanged && fullLead.leadServices.length > 0) {
+      queries.push("UPDATE leads SET modifiedBy = ? WHERE id = ?;");
+      queryData.push(userId, leadId);
+
+      const oldLeadServices = [...oldLead.leadServices];
+      const oldLeadServiceIds = oldLeadServices.map(service => {
+        return service.id;
+      });
+      const fullLeadServiceIds = [...fullLead.leadServices];
```

Comment:
The `fullLeadServiceIds` array does not need to exist. You can just use `fullLead.leadServices`